### PR TITLE
chore: use eslint

### PR DESCRIPTION
**Pull request by agusaldasoro**
This PR fixes https://github.com/decentraland/catalyst/issues/221

## Changes
- Removed `tslinter` in order to use `eslinter`, as the first one was deprecated.
- Configured `eslinter` to run with `prettier` as a dependency. This means that when running `eslinter`, `prettier` will be executed first.
- Configured the project to run `eslinter` as a pre-commit hook.
- Configured the project so if using VS Code IDE, then the linter will run on save.
- Configured the CI to run `yarn lint:check` as a dependency when checking out, to avoid errors by committing code with `--no-verify` flag.

### Note
Please check that I've disabled some validations for a few entire files inside the `comms` folder, as those errors were repeated.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -61,6 +61,8 @@ jobs:
             # fallback to using the latest cache if no exact match is found
             - v1-dependencies-
 
+      - run: yarn lint:check
```
Comment (nchamo) [inline .circleci/config.yml:64]: This probably shouldn't be a part of the checkout. It should be its own job, and run between checkout and other `builds-*`

File: .eslintignore [hunk 2]
```diff
@@ -0,0 +1,2 @@
+comms/peer/src/Peer.ts
```
Comment (nchamo) [inline .eslintignore:1]: How can we fix this file and avoid the need to ignore it?

File: comms/lighthouse/src/handlers.ts [hunk 3]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable @typescript-eslint/ban-types */
```
Comment (nchamo) [inline comms/lighthouse/src/handlers.ts:1]: What does this do?

File: comms/lighthouse/src/server.ts [hunk 4]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable @typescript-eslint/ban-ts-comment */
```
Comment (nchamo) [inline comms/lighthouse/src/server.ts:1]: What does this do?

**Review by nchamo — COMMENTED** [review]
Just a few questions

File: comms/lighthouse/src/handlers.ts (hunk 3 above)
Comment (agusaldasoro) [reply comms/lighthouse/src/handlers.ts:1]: This comments avoids `eslint` to fail when a value is cast to object

File: comms/lighthouse/src/server.ts (hunk 4 above)
Comment (agusaldasoro) [reply comms/lighthouse/src/server.ts:1]: This comments avoids `eslint` to fail when the annotation `@ts-ignore` is used

**Review by nchamo — APPROVED** [review]
Great work!

**Pull request merged**